.github/workflows/code-quality.yml

```diff
@@ -7,27 +7,52 @@ on:
       - "*.latest"
   pull_request:
   workflow_dispatch:
+    inputs:
+      dbt_adapters_branch:
+          description: "The branch of dbt-adapters to evaluate"
+          type: string
+          required: true
+          default: "main"
+  workflow_call:
+    inputs:
+      dbt_adapters_branch:
+        description: "The branch of dbt-adapters to evaluate"
+        type: string
+        required: true
+        default: "main"
 
 permissions: read-all
 
 # will cancel previous workflows triggered by the same event and for the same ref for PRs or same SHA otherwise
 concurrency:
   group: ${{ github.workflow }}-${{ github.event_name }}-${{ contains(github.event_name, 'pull_request') && github.event.pull_request.head.ref || github.sha }}
   cancel-in-progress: true
-
+env:
+  # set DBT_ADAPTERS_BRANCH to the input value if the event is a workflow_dispatch (workflow_call uses the same event_name),
+  # otherwise use 'main'
+  DBT_ADAPTERS_BRANCH: ${{ github.event_name == 'workflow_dispatch' && inputs.dbt_adapters_branch || 'main' }}
```

Comment:
I can't believe this works like this. Also, do we also need to check for `github.event_name == 'workflow_call'`?

Reply:
It's dumb but ya it's a real thing: https://7tonshark.com/posts/github-actions-ternary-operator/